spec/controllers/lambda_callback/document_proof_result_controller_spec.rb

```diff
@@ -53,11 +53,39 @@
                                 document_result: { success: false,
                                                    exception: '#<Proofer::TimeoutError: ' } }
 
-        proofing_result = document_capture_session.load_proofing_result
+        proofing_result = document_capture_session.load_doc_auth_async_result
         expect(proofing_result.result[:exception]).to start_with('#<Proofer::TimeoutError: ')
       end
     end
 
+    context 'with invalid result_id' do
+      before do
+        request.headers['X-API-AUTH-TOKEN'] = AppConfig.env.document_proof_result_lambda_token
+      end
+
+      it 'returns 404' do
+        post :create, params: {
+          result_id: '0000',
+          document_result: {
+          },
+        }, as: :json
+
+        expect(response.status).to eq 404
+      end
+
+      it 'sends exception to new relic' do
+        expect(NewRelic::Agent).to receive(:notice_error).with(
+          'DocumentProofResult result_id not found',
+        )
+
+        post :create, params: {
+          result_id: '0000',
+          document_result: {
+          },
+        }, as: :json
```

Comment:
I'd put this on one line, the extra closing parens threw me when skimming

```suggestion
          result_id: '0000',
          document_result: {},
        }, as: :json
```